File: All Concepts/Longest Substring Without Repeating Characters.cpp
class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        map<char, int> mp;
        if (s.length() == 1)
        {
            return 1;
        }

        int l = 0, r = 0, ans = 0;
        string sb;
        while (r < s.length())
        {
            cout << sb << endl;
            if (mp[s[r]] == 0)
            {
                sb = sb + s[r];
                mp[s[r]]++;
                r++;
            }
            else
            {
                if (sb.length() >= 1)
                {
                    sb = sb.substr(1, sb.length() - 1);
                }
                cout << sb << endl;
                mp[s[l]]--;
                l++;
            }
            if (sb.length() > ans)
            {
                ans = sb.length();
            }
        }
        return ans;
    }
};

File: GS/Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] cArr = new int[128];
        int mxLen =0,l=0,r=0;

        while(r<s.length()){
            char t = s.charAt(r);
            cArr[t]++;
            while(cArr[t]>1){
                cArr[s.charAt(l)]--;
                l++;
            }
            mxLen = Math.max(mxLen,r-l+1);
            r++;
        }
        return mxLen;
    }
}